src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -1818,30 +1784,6 @@ protected virtual void ensure_package_files_have_compatible_attributes(Chocolate
             _filesService.ensure_compatible_file_attributes(installDirectory, config);
         }
 
-        [Obsolete("This overload is obsolete and will be removed in a future version.")]
-        public virtual void rename_legacy_package_version(ChocolateyConfiguration config, IPackageMetadata installedPackage, ChocolateyPackageInformation pkgInfo)
-            => normalize_package_legacy_folder_name(config, installedPackage, pkgInfo);
-
-        protected virtual void normalize_package_legacy_folder_name(ChocolateyConfiguration config, IPackageMetadata installedPackage, ChocolateyPackageInformation pkgInfo)
-        {

```

Comment:
Don't we still need this? For people upgrading directly from very old versions.

Reply:
IMO I don't think it's worth the hassle to keep this. I did initially try to keep this in play, but there were a lot of weird edge case issues, and it's not something we can reliably test from what I was able to tell when messing with this. Our existing tests for this actually relied on being able to use the SideBySide install functionality in the first place to "pretend" the legacy package install.

Given that, I don't think it's sensible to keep this given we're breaking side-by-side functionality completely and deliberately. 

@gep13?

Reply:
> Our existing tests for this actually relied on being able to use the SideBySide install functionality in the first place to "pretend" the legacy package install.

Can't this be switched to using the same method used for testing uninstalls of multiple side by side versions? Unless I misremembering the code.

Reply:
I am inclined to say that we continue with removing this functionality.  We have done everything that we can to let folks know about the upcoming deprecation of side-by-side, and providing an upgrade path from a very old version of Chocolatey is not something that we support.  The only thing that we would support is the previous major version, which in this case would be 1.x, where this method was already in place, and any required normalization has already been done.